airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/debezium/internals/mssql/MssqlDebeziumStateUtil.java

```diff
@@ -66,16 +59,16 @@ public JsonNode constructInitialDebeziumState(final Properties properties,
       publisher.start(queue);
       while (!publisher.hasClosed()) {
         final ChangeEvent<String, String> event = queue.poll(10, TimeUnit.SECONDS);
-        if (event != null) {
+//        if (event != null) {
```

Comment:
@xiaohansong as far as I can tell, whether or not there was an event, closing the engine saves a state.